docs/api/spec.md

```diff
@@ -155,6 +163,36 @@ On the financial desktop, applications often want to broadcast context to any nu
 ## Resolvers
 Intents functionality is dependent on resolver functionality to map the intent to a specific App.  This will often require end-user input.  Resolution can either be performed by the Desktop Agent (raising UI to pick the desired App for the intent) or by the app launching the intent - in which case the calling App will handle the resolution itself (using the findIntents API below) and then invoke an explicit Intent object.
 
+## App Instances
+Using FDC3 APIs, an app can get a reference to an instance of another app that it has either raised an intent to or recieved an intent or context from.  The `AppInstance` API allows apps to listen for and broadcast context directly.  
+
+### Broadcasting and Listening for Context
+An AppInstance MUST support both *broadcast* and *addContextListener* methods.  Calling *broadcast* on an AppInstance MUST send the context only *to* that instance.  The *addContextListener* method MUST listen for context events only *from* that instance. 
```

Comment:
> *addContextListener* method MUST listen for context events only *from* that instance.

How can the application instance, when using `fdc3.broadcast`, differenciate between broadcasting to a channel, that it has joined previously, and broadcasting to an actual "appInstance listener"?
I understand that the application instance can do that differenciation with the `fdc3.addContextListener` API because the handler does provide a source property, which implies that the new context is not coming from a channel broadcasting. But it is not clear on how this distinction can be done with `fdc3.broadcast`.

Reply:
@pgn-vole instance messages can only be handled/passed via addContextListener and broadcast on the instance.  So, fdc3.broadcast would never get routed to an instance.addContextListener handler and vice-versa.  This is consistent with broadcast/handler behavior on channels.  The only exception being when an app has actually joined a channel - in which case - the channel context is the same as the global broadcast.  I will look amending the spec to make that more clear.  

BTWY - The source property won't provide this differentiation - you will always have a single instance source - whether this is coming from a channel, a direct message from an instance, or a global broadcast.  The distinction is only in where the app sets the listener.

Reply:
Thanks for the explanation. So for an application instance to broadcast a message to any of its listener it must beforehand retrieve its own instance and then broadcast via it like the below?

_Listener:_ 
```js
const chart = await fdc3.raiseIntent('ViewChart');
const instance = await fdc3.getAppInstance(chart.source);

instance.addContextListener("someContext", handler);
```

_"ViewChart" Application:_ 
```js
const instance = await fdc3.getAppInstance(myInstanceId); // myInstanceId is exposed in some way by the desktop agent

instance.broadcast({type: "someContext"}) //broadcast its app instance listeners
fdc3.broadcast({type: "someContext"})// broadcast to currently joined channel
```

Reply:
Not exactly. An app shouldn't broadcast on its own instance - but instead, broadcast on the instance of a subscribing app (retrieved from the intentListener or contextListener). So, in the "ViewChart" example above, you would do this instead:

```js
fdc3.addIntentListener("ViewChart",async (context, source) => {
   const sourceInstance = await fdc3.getAppInstance(source);
   sourceInstance.broadcast({type: "someContext"})
});
```

Reply:
Thanks that makes more sense. I got confused by the section "Context last value cache" which made me assume that there would be a single context per application instance, however it seems that instead there will be a context held for each relation provider/subscriber.

Reply:
We had similar issues with initial comprehension here, due to the slightly unconventional (but perhaps necessary) approach.

@nkolba When you say that (the app should) "broadcast on the instance of a subscribing app (retrieved from the intentListener or contextListener)" how does it do that retrieval? It would have needed to register a listener of some sort for the addition of those listeners in order to retrieve their IDs. That seems too complex to be practical.

Perhaps what @pgn-vole suggests is simpler (retrieving its own AppInstance and broadcasting on that) OR perhaps some object is returned by addIntentListener for that purpose (this does not solve the open case however).